packages/docusaurus-plugin-content-docs/src/docFrontMatter.ts

```diff
@@ -29,13 +34,17 @@ export type DocFrontMatter = {
 const DocFrontMatterSchema = Joi.object<DocFrontMatter>({
   id: Joi.string(),
   title: Joi.string().allow(''), // see https://github.com/facebook/docusaurus/issues/4591#issuecomment-822372398
+  hide_title: Joi.boolean(),
+  hide_table_of_contents: Joi.boolean(),
+  keywords: Joi.array().items(Joi.string().required()),
+  image: Joi.string().uri({allowRelative: false}),
   description: Joi.string().allow(''), // see  https://github.com/facebook/docusaurus/issues/4591#issuecomment-822372398
   slug: Joi.string(),
   sidebar_label: Joi.string(),
-  sidebar_position: Joi.number(),
-  custom_edit_url: Joi.string().allow('', null),
+  sidebar_position: Joi.number().min(0),
```

Comment:
I don't think this is technically required and negative values would also work, but it would probably not harm to forbid this